src/lib/slide-toggle/_slide-toggle-theme.scss

```diff
@@ -2,55 +2,63 @@
 @import '../core/theming/theming';
 
 
-@mixin _md-slide-toggle-checked($palette) {
+@mixin _md-slide-toggle-checked($palette, $thumb-on-hue) {
   // Do not apply the checked colors if the toggle is disabled, because the specificity would be to high for
   // the disabled styles.
   &.md-checked:not(.md-disabled) {
     .md-slide-toggle-thumb {
-      background-color: md-color($palette);
+      background-color: md-color($palette, $thumb-on-hue);
     }
 
     .md-slide-toggle-bar {
-      background-color: md-color($palette, 0.5);
+      background-color: md-color($palette, $thumb-on-hue, 0.5);
     }
   }
 }
 
 // TODO(jelbourn): remove this when the real ripple has been applied to slide-toggle.
-@mixin _md-slide-toggle-ripple($palette, $foreground) {
+@mixin _md-slide-toggle-ripple($palette, $foreground, $thumb-on-hue) {
+
   &.md-slide-toggle-focused {
     &:not(.md-checked) .md-ink-ripple {
       // When the slide-toggle is not checked and it shows its focus indicator, it should use a 12% opacity
       // of black in light themes and 12% of white in dark themes.
       background-color: md-color($foreground, divider);
     }
-  }
 
-  &.md-slide-toggle-focused .md-ink-ripple {
-    background-color: md-color($palette, 0.26);
+    .md-ink-ripple {
+      background-color: md-color($palette, $thumb-on-hue, 0.26);
+    }
   }
+
 }
 
 @mixin md-slide-toggle-theme($theme) {
+  $is-dark: map_get($theme, is-dark);
   $primary: map-get($theme, primary);
   $accent: map-get($theme, accent);
   $warn: map-get($theme, warn);
   $background: map-get($theme, background);
   $foreground: map-get($theme, foreground);
 
+  $thumb-on-hue: if($is-dark, 200, 500);
```

Comment:
Can you add a comment here with a link to the spec and explaining the use of different hues based on light vs. dark theme?